api/src/main/java/org/openmrs/api/db/hibernate/HibernateOrderDAO.java

```diff
@@ -647,7 +647,13 @@ public OrderType getOrderTypeByUuid(String uuid) {
 		return (OrderType) sessionFactory.getCurrentSession().createQuery("from OrderType o where o.uuid = :uuid")
 		        .setString("uuid", uuid).uniqueResult();
 	}
-	
+
+	@Override
+	public OrderType getOrderTypeByClassName(String javaClassName) {
+		return sessionFactory.getCurrentSession().createQuery("from OrderType o where o.javaClassName = :javaClassName", OrderType.class)
+			.setMaxResults(1).setString("javaClassName", javaClassName).uniqueResult();
```

Comment:
Do you mind explaining the need for `setMaxResults`?

Reply:
Before adding .`setMaxResults` it was throwing `org.hibernate.NonUniqueResultException: query did not return a unique result: 2`

Searching around for the solution , was to add  `setMaxResults`

Reply:
So does that mean you are intentionally ignoring the second matching result?

Reply:
@dkayiwa yes , because we expect one unique result (Class name) per transaction. Do you think there is a different work  around.

Reply:
That sounds fundamentally wrong. The code should not discard user data, just like that. If the business rule allows more than one record with the same exact class, then it is not correct to search basing on the class name alone.

Reply:
@dkayiwa  if i got this right , do you then suggest having the method search for something generic ?

Reply:
What i mean is that if it was designed in such a way that you are allowed to have more than one record with the same class name, then querying with a class name and expect one result is wrong.

Reply:
ooh ha ha , i understand now , since we're not specifying an individual as is the case with the uuid , then we cannot limit the result set size.